### ISSUE TITLE
minikube kernel version is not able to run IPsec

### ISSUE THREAD

**Issue by nebril** [cilium/cilium#15962]
Current minikube version is `4.19.114`. Kernel config is here: https://github.com/kubernetes/minikube/blob/v1.19.0/deploy/iso/minikube-iso/board/coreos/minikube/linux_defconfig

When trying to run Cilium with IPsec enabled on minikube cluster, Cilium pod fails with following log message:

```
level=fatal msg="IPSec with tunneling requires support for xfrm state output masks (Linux 4.19 or later)." error="protocol not supported" subsys=daemon
```

@pchaigno suggested that there might be a commit between `4.19.114` (minikube) and `4.19.177` (our CI) that is required for IPsec support.

**Comment by pchaigno** [comment]
I had a look at xfrm commits between these two patch versions with `git log --pretty=oneline v4.19.114..v4.19.177 | grep xfrm`. The only one I found that could maybe be related is:
```
commit 50e117921b322323b7272f108d9c080ad883ee0a
Author: Antony Antony <antony.antony@secunet.com>
Date:   Fri Sep 4 08:49:38 2020 +0200

    xfrm: clone XFRMA_SET_MARK in xfrm_do_migrate
    
    [ Upstream commit 545e5c571662b1cd79d9588f9d3b6e36985b8007 ]
    
    XFRMA_SET_MARK and XFRMA_SET_MARK_MASK was not cloned from the old
    to the new. Migrate these two attributes during XFRMA_MSG_MIGRATE
    
    Fixes: 9b42c1f179a6 ("xfrm: Extend the output_mark to support input direction and masking.")
    Signed-off-by: Antony Antony <antony.antony@secunet.com>
    Signed-off-by: Steffen Klassert <steffen.klassert@secunet.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>
```

**Comment by pchaigno** [comment]
IPsec is failing on Minikube because the kernel used when running in a VM is missing a lot of configs required to run Cilium's IPsec implementation. The full list of required configs can be seen at https://github.com/cilium/packer-ci-build/blob/407062ec454ab38ecdf1399bc3176df7f54fb182/provision/ubuntu/kernel-next.sh#L109-L131 and will be documented as part of https://github.com/cilium/cilium/issues/14852.
